src/main/java/com/ubc/cpsc304/hotelify/service/HotelService.java

```diff
@@ -0,0 +1,76 @@
+package com.ubc.cpsc304.hotelify.service;
+
+import com.ubc.cpsc304.hotelify.controller.dto.HotelRequestDto;
+import com.ubc.cpsc304.hotelify.entity.Hotel;
+import com.ubc.cpsc304.hotelify.exception.BadRequestException;
+import com.ubc.cpsc304.hotelify.repository.HotelRepository;
+import org.hibernate.annotations.NotFound;
+import org.hibernate.annotations.NotFoundAction;
+import org.springframework.stereotype.Service;
+
+import javax.persistence.EntityNotFoundException;
+
+/**
+ * Service layer for com.ubc.cpsc304.hotelify.entity.Hotel
+ * Scott 2018-11-05
+ */
+
+@Service
+public class HotelService {
+    private final HotelRepository hotelRepository;
+
+    public HotelService(HotelRepository hotelRepository) {
+        this.hotelRepository = hotelRepository;
+    }
+
+    public Hotel createHotel(HotelRequestDto requestDto) throws BadRequestException{
+        String brandName = requestDto.getBrandName();
+        Long addressId = requestDto.getAddressId();
+
+        if(brandName==null || addressId==null){
+            throw new BadRequestException("brandName and addressId cannot be null");
+        }
+
+        Hotel hotelToInsert = new Hotel();
+        hotelToInsert.setBrandName(requestDto.getBrandName());
+        hotelToInsert.setBranchName(requestDto.getBranchName());
+        hotelToInsert.setPropertyClass(requestDto.getPropertyClass());
+        hotelToInsert.setAddressId(requestDto.getAddressId());
+        hotelToInsert.setDescription(requestDto.getDescription());
+        hotelToInsert.setOverallRating(requestDto.getOverallRating());
+
+        return this.hotelRepository.save(hotelToInsert);
+    }
+
+    public Hotel updateHotelById(HotelRequestDto requestDto) throws BadRequestException{
```

Comment:
if the method name is `xxxxById`, then the parameter passed in should at least include the "Id“